soco/core.py

```diff
@@ -1501,16 +1501,32 @@ def _music_lib_search(self, search, start, max_items):
             metadata[camel_to_underscore(tag)] = int(response[tag])
         return response, metadata
 
-    def add_uri_to_queue(self, uri):
+    def add_uri_to_queue(self, uri, meta=''):
         """Adds the URI to the queue
 
         :param uri: The URI to be added to the queue
         :type uri: str
+        :param meta: The track metadata to show in the player, DIDL format.
+        :type uri: str
         """
         # FIXME: The res.protocol_info should probably represent the mime type
         # etc of the uri. But this seems OK.
         res = [DidlResource(uri=uri, protocol_info="x-rincon-playlist:*:*:*")]
-        item = DidlObject(resources=res, title='', parent_id='', item_id='')
+        if meta == '':
+            item = DidlObject(resources=res,
+                              title='', parent_id='', item_id='')
+        else:
+            didl_objects = from_didl_string(meta)
+            if didl_objects is None or len(didl_objects) == 0:
+                _LOG.debug('Failed to construct metadata object'
+                           'from string')
+                return False
```

Comment:
We are trying to get away from using the return value to indicate success. I would suggest to raise a ValueError here instead.